### ISSUE TITLE
Factor and CompositeFactor not working

### ISSUE THREAD

**Issue by rcichota** [APSIMInitiative/ApsimX#3790]
This could be two issues, maybe, but then they may be related, and be related to changes being made in #3716 
In an experiment, one can drop a factor, using add model (mouse click menu), and then add CompositeFactor under it in the same way. Doing this one cannot set instructions in CompositeFactor, that is, one can write the instruction but it vanishes as soon as one click somewhere else and then back... The weird thing is that if one copies a working CompositeFactor from another simulation and then do the appropriate modifications it works!!
The related problem is on Factor. If the text box is edited, either by adding an instruction or by mistake, and then the user decides to delete (say because one wants to use CompositeFactor), the simulation get stuck in a error ('cannot find model'). I traced this to the fact that the variable ```Specification``` (in Factor.cs), which holds the info from the text box, is *empty* but not *null*. If the text box in Factor is never edited that variable is always evaluated as null, but this is not the case for after deleting the contents, it returns an empty string. Apsim then attempts to find that "model" but always fails...

**Comment by rcichota** [comment]
Some background on how i got to this issue:
I am trying to build a simple sensitivity analysis for white clover and I want to test a few parameters/functions. I have it working by setting up a factor and change one function I want to test, e.g.:

![image](https://user-images.githubusercontent.com/12175986/57005832-3a5dcf80-6c2f-11e9-974e-56da78eac53c.png)

However, because I am changing things within the model, the instruction is complex and has square brackets. Using simply Factor results in rather ugly simulation and then factor names. I thought of using sub-factors, but that doesn't seem to be allowed, unless one uses CompositeFactor. I did that and now I have something working and giving me nice factor names to use in graphs etc:

![image](https://user-images.githubusercontent.com/12175986/57005812-0f737b80-6c2f-11e9-899c-779388ce7c88.png)

But to get to this I needed to copy the structure from another working simulation (I used AgPasture's) and modify it, if I try to do it from scratch it doesn't work. And the error messages are no help in identifying the issue (esepcially not being able to edit CompositeFactor)...
